controller/src/main/java/io/pravega/controller/store/stream/Segment.java

```diff
@@ -15,19 +15,21 @@
 import lombok.ToString;
 
 import java.util.AbstractMap;
+import lombok.NoArgsConstructor;
 
 /**
  * Properties of a stream segment that don't change over its lifetime.
  */
 @Data
+@NoArgsConstructor
```

Comment:
We don't want a no arg constructor. A segment with everthing set to null is an invalid object.